server/channels/api4/user_test.go

```diff
@@ -7944,3 +7946,580 @@ func TestLoginWithDesktopToken(t *testing.T) {
+	})
+
+	t.Run("set unread state for different user", func(t *testing.T) {
+		// Log back in
+		client.Login(context.Background(), user.Email, user.Password)
+
+		post, _, err := client.CreatePost(context.Background(), &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			Message:   "Root message for different user test",
+		})
+		require.NoError(t, err)
+
+		// Try to set unread state for a different user
+		_, resp, err := client.SetThreadUnreadByPostId(context.Background(), th.BasicUser2.Id, team.Id, post.Id, post.Id)
+		require.Error(t, err)
+		CheckForbiddenStatus(t, resp)
+	})
+
+	t.Run("set unread state as system admin", func(t *testing.T) {
+		post, _, err := th.SystemAdminClient.CreatePost(context.Background(), &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			Message:   "Root message for system admin test",
+		})
+		require.NoError(t, err)
+
+		reply, _, err := th.SystemAdminClient.CreatePost(context.Background(), &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			RootId:    post.Id,
+			Message:   "Reply for system admin test",
+		})
+		require.NoError(t, err)
+
+		thread, resp, err := th.SystemAdminClient.SetThreadUnreadByPostId(context.Background(), user.Id, team.Id, post.Id, reply.Id)
+		require.NoError(t, err)
+		CheckOKStatus(t, resp)
+		require.NotNil(t, thread)
+		require.Equal(t, post.Id, thread.PostId)
+
+		// Check that LastReplyAt is a recent timestamp
+		require.Greater(t, thread.LastReplyAt, int64(0))
+		require.LessOrEqual(t, thread.LastReplyAt, model.GetMillis())
+
+		// Check if the thread is marked as unread
+		require.True(t, thread.UnreadReplies > 0, "Thread should have unread replies")
+
+		require.InDelta(t, model.GetMillis(), thread.LastReplyAt, float64(5000), "LastReplyAt should be within 5 seconds of current time")
+	})
+}
+
+func TestRevokeAllSessionsForUser(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	user := th.BasicUser
+	user2 := th.BasicUser2 // Additional user for permission testing
+
+	// Create multiple sessions for the primary user
+	client1 := th.CreateClient()
+	_, _, err := client1.Login(context.Background(), user.Email, user.Password)
+	require.NoError(t, err)
+
+	client2 := th.CreateClient()
+	_, _, err = client2.Login(context.Background(), user.Email, user.Password)
+	require.NoError(t, err)
+
+	// Create a session for the second user (non-admin)
+	nonAdminClient := th.CreateClient()
+	_, _, err = nonAdminClient.Login(context.Background(), user2.Email, user2.Password)
+	require.NoError(t, err)
+
+	t.Run("Revoke all sessions as admin", func(t *testing.T) {
+		resp, err := th.SystemAdminClient.RevokeAllSessions(context.Background(), user.Id)
+		require.NoError(t, err)
+		CheckOKStatus(t, resp)
+
+		// Use SystemAdminClient to verify that all sessions are revoked
+		sessions, _, err := th.SystemAdminClient.GetSessions(context.Background(), user.Id, "")
+		require.NoError(t, err)
+		require.Empty(t, sessions, "All sessions should be revoked")
+	})
```

Comment:
On my previous review I recommended including `LocalClient` where applicable. Is this something the AI model can handle?

```suggestion
	th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
		resp, err := th.SystemAdminClient.RevokeAllSessions(context.Background(), user.Id)
		require.NoError(t, err)
		CheckOKStatus(t, resp)

		// Use SystemAdminClient to verify that all sessions are revoked
		sessions, _, err := th.SystemAdminClient.GetSessions(context.Background(), user.Id, "")
		require.NoError(t, err)
		require.Empty(t, sessions, "All sessions should be revoked")
	}, "Revoke all sessions as admin and local")
```